packages/bsky/src/db/index.ts

```diff
@@ -46,11 +56,21 @@ export class Database {
   }
 
   async transaction<T>(fn: (db: Database) => Promise<T>): Promise<T> {
-    const res = await this.db.transaction().execute(async (txn) => {
-      const dbTxn = new Database(txn, this.cfg)
-      const txRes = await fn(dbTxn)
-      return txRes
-    })
+    let res: Awaited<T>
+    const leakyTxPlugin = new LeakyTxPlugin()
+    try {
+      res = await this.db
+        .withPlugin(leakyTxPlugin)
+        .transaction()
+        .execute(async (txn) => {
+          const dbTxn = new Database(txn, this.cfg)
+          const txRes = await fn(dbTxn)
+          return txRes
+        })
+    } catch (err) {
+      leakyTxPlugin.endTx()
```

Comment:
To fully avoid a race I wonder if we should end the transaction immediately before the commit/rollback (and in both cases), i.e. as soon as we know `fn` should be done with it.  What do you think?

Reply:
Could look like:
```ts
const txRes = await fn(dbTxn)
  .finally(() => leakyTxPlugin.endTx())
```